### ISSUE TITLE
`curl_easy_getinfo` type checking failed

### ISSUE THREAD

**Issue by Billyzou0741326** [curl/curl#5432]
I was using the `CURLOPT_PRIVATE` to store some data. Here's what I did,

```c
struct http_request_private_data *priv_data;
curl_easy_getinfo(easy_curl, CURLINFO_PRIVATE, &priv_data);
```

But then when I compiled with Linux gcc, I got this error (as in issue #846),

```
curl/typecheck-gcc.h:122:13: error: call to ‘_curl_easy_getinfo_err_string’ declared with attribute warning: curl_easy_getinfo expects a pointer to 'char *' for this info [-Werror]
             _curl_easy_getinfo_err_string();
```

Curiously, there was no warning when I compiled with `-O0` flag, without making any changes to the code.
When using any other optimization flags, `-O2` for instance, the warning shows up.

### curl/libcurl version

curl-config --version
`libcurl 7.70.0`

### operating system

Linux 4.15.0-91-generic #92-Ubuntu SMP Fri Feb 28 11:09:48 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux

**Comment by bagder** [comment]
Ack. `CURLINFO_PRIVATE` is set to have a `CURLINFO_STRING` type which then leads to that. We can't change the number now but we can add an exception to `typecheck-gcc.h`.

Thanks for this report!

**Comment by bagder** [comment]
Uh, why close it?

**Comment by bagder** [comment]
@Billyzou0741326 can you check if the fix in #5436 is fine for you?

**Comment by minamiktr** [comment]
Yes, compiled successfully. Thank you!
